recipes/opencv/4.x/conanfile.py

```diff
@@ -0,0 +1,395 @@
+            return "%s%s%s%s" % (prefix, module, version, debug)
+
+        def add_components(components):
+            # TODO: OpenCV doesn't use cmake target namespace
+            for component in components:
+                conan_component = component["target"]
+                cmake_target = component["target"]
+                lib_name = get_lib_name(component["lib"])
+                requires = component["requires"]
+                self.cpp_info.components[conan_component].names["cmake_find_package"] = cmake_target
+                self.cpp_info.components[conan_component].names["cmake_find_package_multi"] = cmake_target
+                self.cpp_info.components[conan_component].libs = [lib_name]
+                self.cpp_info.components[conan_component].includedirs.append(os.path.join("include", "opencv4"))
+                self.cpp_info.components[conan_component].requires = requires
+                if self.settings.os == "Linux":
+                    self.cpp_info.components[conan_component].system_libs = ["dl", "m", "pthread", "rt"]
+
+                # CMake components names
+                conan_component_alias = conan_component + "_alias"
+                cmake_component = component["lib"]
+                self.cpp_info.components[conan_component_alias].names["cmake_find_package"] = cmake_component
+                self.cpp_info.components[conan_component_alias].names["cmake_find_package_multi"] = cmake_component
+                self.cpp_info.components[conan_component_alias].requires = [conan_component]
+                self.cpp_info.components[conan_component_alias].includedirs.append(os.path.join("include", "opencv4"))
+                self.cpp_info.components[conan_component_alias].libdirs = []
+                self.cpp_info.components[conan_component_alias].resdirs = []
+                self.cpp_info.components[conan_component_alias].bindirs = []
+                self.cpp_info.components[conan_component_alias].frameworkdirs = []
+
+        def imageformats_deps():
+            components = []
+            if self.options.with_jpeg2000:
+                components.append("{0}::{0}".format(self.options.with_jpeg2000))
+            if self.options.with_png:
+                components.append("libpng::libpng")
+            if self.options.with_jpeg:
+                components.append("{0}::{0}".format(self.options.with_jpeg))
+            if self.options.with_tiff:
+                components.append("libtiff::libtiff")
+            if self.options.with_openexr:
+                components.append("openexr::openexr")
+            if self.options.with_webp:
+                components.append("libwebp::libwebp")
+            return components
+
+        def eigen():
+            return ["eigen::eigen"] if self.options.with_eigen else []
+
+        def parallel():
+            if self.options.parallel:
+                return ["tbb::tbb"] if self.options.parallel == "tbb" else ["openmp"]
```

Comment:
How does openmp work? we do not have a `self.requires("...")` for openmp but we add it `self.cpp_info.components[conan_component].requires = requires`